jdbc/sqlserver/src/main/java/com/google/cloud/sql/sqlserver/SocketFactory.java

```diff
@@ -36,8 +37,16 @@ public class SocketFactory extends javax.net.SocketFactory {
    * Implements the {@link SocketFactory} constructor, which can be used to create authenticated
    * connections to a Cloud SQL instance.
    */
-  public SocketFactory(String instanceName) {
-    this.props.setProperty(CoreSocketFactory.CLOUD_SQL_INSTANCE_PROPERTY, instanceName);
+  public SocketFactory(String socketFactoryConstructorArg) {
+    String[] s = socketFactoryConstructorArg.split("\\?");
```

Comment:
You may want to use String.indexOf('?') instead of String.split(). Customer may include more than one '?' characters and you don't really want to split on all '?', just the first one.

Reply:
What would be the use case for more than one `?` character? Wouldn't that be a malformed string (which we should probably catch with an error). so far the only allowed args are true/false for `enableIamAuth` and public/private for ipType.

Reply:
Can we use `https://docs.oracle.com/javase/8/docs/api/java/net/URI.html`?

Reply:
> Can we use [ParseURI](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html)?


Unfortunately not - I tried it and the `:` character in the instance name causes issues, and we can't make users escape the instance name without breaking existing users (see check failures in [53b5936](https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/pull/936/commits/53b5936e5fd56f03fc0fb6f9dceeb9d2b2bf63cc))